internal/services/nginx/nginx_deployment_resource.go

```diff
@@ -179,7 +189,37 @@ func (m DeploymentResource) Arguments() map[string]*pluginsdk.Schema {
 			},
 		},
 
+		"scaling": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: false,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"capacity": {
+						Type:         pluginsdk.TypeInt,
+						Required:     true,
+						ValidateFunc: validation.IntPositive,
+					},
+				},
+			},
+		},
+
 		"tags": commonschema.Tags(),
+
+		"user_profile": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: false,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"preferred_email": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
```

Comment:
Given there is only one property in this block, I think it would make sense to flatten this and make it a top-level property called `email` - this would also make it consistent with other resources that have a field for a support email address.

Reply:
Checking with team to see if there are plans to expand this API, but think it is safe to say not in the short term and we can consolidate this property.

Reply:
Made changes to flatten schema for this and scaling.